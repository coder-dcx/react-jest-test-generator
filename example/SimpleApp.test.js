import React from 'react';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

// Configure Enzyme adapter
configure({ adapter: new Adapter() });

import { SimpleComponent, AnotherComponent } from './SimpleApp';

describe('SimpleComponent', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SimpleComponent />);
    expect(wrapper.exists()).toBe(true);
  });

  it('renders with custom title', () => {
    const wrapper = shallow(<SimpleComponent title="Test Title" />);
    expect(wrapper.find('h1').text()).toBe('Test Title');
  });

  it('displays initial count', () => {
    const wrapper = shallow(<SimpleComponent />);
    expect(wrapper.find('p').text()).toBe('Count: 0');
  });

  it('increments count when button is clicked', () => {
    const wrapper = shallow(<SimpleComponent />);
    const button = wrapper.find('button').at(0); // First button (Increment)
    button.simulate('click');
    expect(wrapper.find('p').text()).toBe('Count: 1');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<SimpleComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('AnotherComponent', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<AnotherComponent />);
    expect(wrapper.exists()).toBe(true);
  });

  it('renders with custom name', () => {
    const wrapper = shallow(<AnotherComponent name="John" />);
    expect(wrapper.text()).toBe('Hello John');
  });

  it('renders with default name', () => {
    const wrapper = shallow(<AnotherComponent />);
    expect(wrapper.text()).toBe('Hello World');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<AnotherComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});