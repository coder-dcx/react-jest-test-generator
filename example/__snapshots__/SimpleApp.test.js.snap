// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AnotherComponent matches snapshot 1`] = `ShallowWrapper {}`;

exports[`SimpleComponent matches snapshot 1`] = `ShallowWrapper {}`;
